pkgs/os-specific/darwin/moltenvk/default.nix

```diff
@@ -49,106 +59,113 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-0+S/kueV+AEVt+oFnh4cgcDRVtEbUH1QiHFPhGhimCA=";
   };
 
-  patches = [
-    # Fix the Xcode projects to play nicely with `xcbuild`.
-    ./MoltenVKShaderConverter.xcodeproj.patch
-    ./MoltenVK.xcodeproj.patch
-  ];
-
   postPatch = ''
     # Move `mvkGitRevDerived.h` to a stable location
     substituteInPlace Scripts/gen_moltenvk_rev_hdr.sh \
       --replace-fail '$'''{BUILT_PRODUCTS_DIR}' "$NIX_BUILD_TOP/$sourceRoot/build/include" \
       --replace-fail '$(git rev-parse HEAD)' ${finalAttrs.src.rev}
-    # Use the SPIRV-Cross packaged in nixpkgs instead of one built specifically for MoltenVK.
-    substituteInPlace MoltenVK/MoltenVK.xcodeproj/project.pbxproj \
-      --replace-fail SPIRV_CROSS_NAMESPACE_OVERRIDE=MVK_spirv_cross SPIRV_CROSS_NAMESPACE_OVERRIDE=spirv_cross
+
+    # Modify MoltenVK Xcode projects to build with xcbuild and dependencies from nixpkgs
+    for proj in MoltenVK MoltenVKShaderConverter; do
+      sed -E -i "$proj/$proj.xcodeproj/project.pbxproj" -e '/(children|files) = /,/;/{/xcframework/d}'
```

Comment:
this sed expression deletes lines that contain `xcframework` that come between `children|files =` ... `;`?

for constancy keep the args and flags for the sed in the same order across the file.

Reply:
I can update to make the order consistent. I can also make it clear why it’s deleting those lines.

MoltenVK builds its dependencies as xcframeworks then references them in the project. Aside from wanting to use dependencies from nixpkgs instead, we have no way of building xcframeworks. Previously, the MoltenVK derivation patched out the xcframework dependencies, but the Xcode projects have changed the last few versions (invalidating the patches). I replaced the patches with a script to avoid having to recreate the patch every release.

The basic format of this section looks like:

```
files = (
    DCFD7F812A45BDA0007BBBF7 /* SPIRVTools.xcframework in Frameworks */,
    DCFD7F822A45BDA0007BBBF7 /* SPIRVCross.xcframework in Frameworks */,
    DCFD7F832A45BDA0007BBBF7 /* glslang.xcframework in Frameworks */,
);
```

The sed script is deleting lines with xcframeworks that are found in the address ranges corresponding to those delimiters. I’m not deleting them unconditionally because I don’t know what else mentions them and don’t want to risk corrupting the file just to remove a few dependencies.

Reply:
maybe just a comment like this is sufficient.
`# remove lines tagged with xcframework located in files or child blocks`

Reply:
I added a slightly more verbose comment, but this should be done now.